### PR TITLE
chore: produce core files in regtests

### PR DESCRIPTION
**Pull request by romange**
Should work only for self-hosted runners.
The core files will be kept in /var/crash/
We also copy automatically dragonfly binary into /var/crash to be able to debug later.

### CODE REVIEW

File: tests/dragonfly/requirements.txt [hunk 1]
```diff
@@ -15,7 +15,7 @@ pytest-repeat==0.9.3
 pymemcache==4.0.0
 prometheus_client==0.17.0
 aiohttp==3.10.2
-numpy==1.24.3
+numpy
```
Comment (kostasrim) [inline tests/dragonfly/requirements.txt:18]: Why this change?

File: .github/workflows/regression-tests.yml [hunk 2]
```diff
@@ -56,6 +60,16 @@ jobs:
           name: logs
           path: /tmp/failed/*
 
+      - name: Copy binary on a self hosted runner
+        if: failure()
+        run: |
```
Comment (kostasrim) [inline .github/workflows/regression-tests.yml:65]: Not sure if you "tested" this. I can inject a failure to make sure that it works if you can wait until tomorrow.

**Review by kostasrim — APPROVED** [review]

File: tests/dragonfly/requirements.txt (hunk 1 above)
Comment (romange) [reply tests/dragonfly/requirements.txt:18]: because numpy is better to install via apt and its version depends on each linux distribution.
these tests do not rely on a specific version

File: .github/workflows/regression-tests.yml (hunk 2 above)
Comment (romange) [reply .github/workflows/regression-tests.yml:65]: i tested this.
